sys/include/xtimer/tick_conversion.h

```diff
@@ -12,9 +13,56 @@
  * @{
  * @file
  * @brief   xtimer tick <-> seconds conversions for different values of XTIMER_HZ
+ *
  * @author  Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ * @author  Josua Arndt <jarndt@ias.rwth-aachen.de>
  */
 
+/*
+ * As tick_conversion.h is included in multiple files ENABLE_DEBUG can not be
+ * used as it gets redefined "[-Werror]"
+ * To use  the debug print enable module fmt in the project makefile
+ *
+ * USEMODULE += fmt
+ * CFLAGS += -DENABLE_DEBUG_TICKS=1
+ * */
+#ifndef ENABLE_DEBUG_TICKS
+#define ENABLE_DEBUG_TICKS   (0)
+#endif
+#if ENABLE_DEBUG_TICKS
+#ifndef MODULE_FMT
+#error "Module fmt is necessary to ENABLE_DEBUG_TICKS set 'USEMODULE += fmt' \
+in the makefile."
+#endif
+#include <stdio.h>
+#include <string.h>
+#include "fmt.h"
+#define DEBUG_PRINT_TICKS_32( CLOCK_HZ, EQUAL, VALUE1, VALUE2, BASE2) ({\
+char buf_1[21];                                                         \
+char buf_2[21];                                                         \
+size_t size = fmt_u32_dec(buf_1, VALUE1);                               \
+buf_1[size] = '\0';                                                     \
+size = fmt_u32_dec(buf_2, VALUE2);                                      \
+buf_2[size] = '\0';                                                     \
+printf("32bit %" PRIu32 " "EQUAL" 1MHz : %s "#VALUE1" = %s "BASE2"\n",  \
+       (uint32_t)CLOCK_HZ , buf_1, buf_2);                              \
+})
+
+#define DEBUG_PRINT_TICKS_64( CLOCK_HZ, EQUAL, VALUE1, VALUE2, BASE2) ({\
+char buf_1[21];                                                         \
+char buf_2[21];                                                         \
+size_t size = fmt_u64_dec(buf_1, VALUE1);                               \
+buf_1[size] = '\0';                                                     \
+size = fmt_u64_dec(buf_2, VALUE2);                                      \
+buf_2[size] = '\0';                                                     \
+printf("64bit %" PRIu32 " "EQUAL" 1MHz : %s "#VALUE1" = %s "BASE2"\n",  \
+       (uint32_t)CLOCK_HZ , buf_1, buf_2);                              \
+})
+#else
+#define DEBUG_PRINT_TICKS_32(CLOCK_HZ, EQUAL, VALUE1, VALUE2, BASE2) {}
+#define DEBUG_PRINT_TICKS_64(CLOCK_HZ, EQUAL, VALUE1, VALUE2, BASE2) {}
```

Comment:
We have reached a consensus regarding ENABLE_DEBUG to use normal if statements and rely on the optimizer to discard the dead code instead of discarding the code in the preprocessor. The benefit is that the debug prints are checked for syntax errors and mistakes by the CI system even without enabling debug.
The same approach should be applicable here too.